packages/nextjs/components/NetworksDropdown.tsx

```diff
@@ -80,36 +110,104 @@ export const NetworksDropdown = ({ onChange }: { onChange: (options: any) => any
     }
   }, []);
 
+  const handleSelectChange = (newValue: SingleValue<Options> | MultiValue<Options>) => {
+    const selected = newValue as SingleValue<Options>;
+    if (selected?.value === "see-all") {
+      if (seeAllModalRef.current) {
+        seeAllModalRef.current.showModal();
+      }
+      if (searchInputRef.current) {
+        searchInputRef.current.focus();
+      }
+    } else {
+      setSelectedOption(selected);
+      onChange(selected);
+    }
+  };
+
+  const filteredChains = allChains.filter(chain => chain.label.toLowerCase().includes(searchTerm.toLowerCase()));
```

Comment:
Should we filter also by the Chain Id in case some user searches by that?

```suggestion
  const filteredChains = allChains.filter(
    chain =>
      chain.label.toLowerCase().includes(searchTerm.toLowerCase()) ||
      chain.value.toString().toLowerCase().includes(searchTerm.toLowerCase()),
  );
  ```